zzre/game/systems/ui/ScrBookMenu.cs

```diff
@@ -42,27 +69,111 @@ protected override void HandleOpen(in messages.ui.OpenBookMenu message)
             .Build();
 
         CreateTopButtons(preload, entity, inventory, IDOpenFairybook);
-        CreateFairyButtons(preload, entity, inventory);
+        CreateFairyButtons(preload, entity, inventory, ref book);
     }
 
-    private void CreateFairyButtons(UIPreloader preload, in DefaultEcs.Entity entity, Inventory inventory)
+    private void CreateFairyButtons(UIPreloader preload, in DefaultEcs.Entity entity, Inventory inventory, ref components.ui.ScrBookMenu book)
     {
-        var fairies = db.Fairies.OrderBy(fairyRow => fairyRow.CardId.EntityId).ToArray();
+        var fairies = book.Fairies;
         for (int i = 0; i < fairies.Length; i++)
         {
             if (inventory.Contains(fairies[i].CardId))
             {
                 var element = new components.ui.ElementId(1 + i);
-                preload.CreateButton(entity)
+                var button = preload.CreateButton(entity)
                     .With(element)
-                    .With(Mid + new Vector2(226 + 45 * (i % 9), 66 + 45 * (i / 9)))
+                    .With(Mid + FairyButtonPos(i))
                     .With(new components.ui.ButtonTiles(fairies[i].CardId.EntityId))
                     .With(preload.Wiz000)
                     .Build();
+                book.FairyButtons.Add(element, fairies[i]);
+
+                // If player has multiple fairies of the same card id, only the first caught counts
+                if (inventory.Fairies.First(c => fairies[i].CardId == c.cardId).isInUse) {
```

Comment:
Personally I don't care much about this bug, like the instant ScrMapMenu fade it is a bit too obscure to care about. What do you think?